pkg/controller/networkpolicy/networkpolicy_controller.go

```diff
@@ -637,8 +637,21 @@ func (n *NetworkPolicyController) updateNetworkPolicy(old, cur interface{}) {
 // deleteNetworkPolicy receives NetworkPolicy DELETED events and deletes resources
 // which can be consumed by agents to delete corresponding rules on the Nodes.
 func (n *NetworkPolicyController) deleteNetworkPolicy(old interface{}) {
+	np, ok := old.(*networkingv1.NetworkPolicy)
+	if !ok {
+		tombstone, ok := old.(cache.DeletedFinalStateUnknown)
+		if !ok {
+			klog.Errorf("error decoding object %v, invalid type", old)
```

Comment:
maybe add more info here and below .. like error decoding NetworkPolicy object while delete phase, or something like that?